Zed quit unexpectedly while opening new directory

### Check for existing issues

- [x] Completed

### Describe the bug / provide steps to reproduce it

Zed quit unexpectedly while opening new directory

### Environment

Zed version: 0.169.361b01a0ca91942dcaf59161c6973b77c617f5238
OS: macOS 15.2.0
Architecture:  aarch64
RAM: 64Gb

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.

<details><summary>Zed.log</summary>

<!-- Click below this line and paste or drag-and-drop your log-->
```
2025-01-22T01:21:58.403862-08:00 [INFO] building git repository, `.git` path in the worktree: ".git"
2025-01-22T01:21:58.432566-08:00 [ERROR] {
  "thread": "<unnamed>",
  "payload": "called `Option::unwrap()` on a `None` value",
  "location_data": {
    "file": "/Users/administrator/.cargo/registry/src/index.crates.io-6f17d22bba15001f/git2-0.19.0/src/remote.rs",
    "line": 150
  },
  "backtrace": [
    "backtrace::capture::Backtrace::create::hbbd36bf6d932e412+8825176",
    "backtrace::capture::Backtrace::new::h1f2c8bb8a22a1648+8824988",
    "zed::reliability::init_panic_hook::{{closure}}::h2a767cb3729bf893+77398476",
    "std::panicking::rust_panic_with_hook::h09e8a656f11e82b2+59102676",
    "std::panicking::begin_panic_handler::{{closure}}::h1230eb3cc91b241c+59099732",
    "std::sys::backtrace::__rust_end_short_backtrace::hc3491307aceda2c2+59092532",
    "_rust_begin_unwind+59098960",
    "core::panicking::panic_fmt::ha4b80a05b9fff47a+84912116",
    "core::panicking::panic::h298549a7412a7069+84912224",
    "core::option::unwrap_failed::hb7af631ec4f78cd6+84911996",
    "<git::repository::RealGitRepository as git::repository::GitRepository>::remote_url::h3f20247c8ec9bd67+27885648",
    "git_hosting_providers::register_additional_providers::haf2fa346e24b5480+27976516",
    "worktree::BackgroundScannerState::insert_git_repository_for_path::h602e84025fc0134e+74923312",
    "worktree::BackgroundScannerState::insert_git_repository::h21047572ef2a5c95+74917892",
    "worktree::BackgroundScanner::scan_dir::{{closure}}::h42f57d781507ab82+74972012",
    "gpui::executor::Scope::spawn::{{closure}}::hac5ba03dfaa0c52a+74567404",
    "async_task::raw::RawTask<F,T,S,M>::run::h5f7a788a51784f26+74360448",
    "gpui::platform::mac::dispatcher::trampoline::h146ce582c8b0d5b2+30334036",
    "_pthread_mach_thread_np+2257077148"
  ],
  "app_version": "0.169.3",
  "release_channel": "stable",
  "target": "aarch64-apple-darwin",
  "os_name": "macOS",
  "os_version": "15.2.0",
  "architecture": "aarch64",
  "panicked_on": 1737537718432,
  "system_id": "31c2208c-be28-4518-852c-8367048d502e",
  "installation_id": "f4699349-5920-460c-b919-ac24b81a0664",
  "session_id": "e1be70f3-9c3a-4290-a96c-e5ec0af7dc86"
}

```
<!-- Click above this line and paste or drag-and-drop your log--></details>

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -400,7 +400,8 @@ fork = "0.2.0"
 futures = "0.3"
 futures-batch = "0.6.1"
 futures-lite = "1.13"
-git2 = { version = "0.19", default-features = false }
+# TODO: get back to regular versions when https://github.com/rust-lang/git2-rs/pull/1120 is released
+git2 = { git = "https://github.com/rust-lang/git2-rs", rev = "a3b90cb3756c1bb63e2317bf9cfa57838178de5c", default-features = false }
 globset = "0.4"
 handlebars = "4.3"
 heed = { version = "0.21.0", features = ["read-txn-no-tls"] }

